### PR TITLE
Fixed duplicate message sending to bots and email after changing order status

### PR DESCRIPTION
**Pull request by MarkAndzh**
# GreenCityUBS PR
Fixed duplicate message sending to bots and email after changing order status

## Summary Of Changes:
Added static NotificationReceiverType to all notification providers. Changed sendNotificationsForBotsAndEmail method. Fixed TelegramApiException.

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

**Review by ABbondar — APPROVED** [review]

**Review by BranetE — APPROVED** [review]

**Review by sergoliarnik — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by MaksymGolik — APPROVED** [review]

File: core/pom.xml [hunk 1]
```diff
@@ -124,7 +124,7 @@
         <dependency>
             <groupId>org.telegram</groupId>
             <artifactId>telegrambots-spring-boot-starter</artifactId>
-            <version>4.9.2</version>
+            <version>5.4.0.1</version>
```
Comment (VadymMakitra) [inline core/pom.xml:127]: Please declare version in properties.

File: service/pom.xml [hunk 2]
```diff
@@ -117,7 +117,7 @@
         <dependency>
             <groupId>org.telegram</groupId>
             <artifactId>telegrambots-spring-boot-starter</artifactId>
-            <version>5.0.1.1</version>
+            <version>5.4.0.1</version>
```
Comment (VadymMakitra) [inline service/pom.xml]: Please declare version in properties.

File: service/src/main/java/greencity/service/notification/AbstractNotificationProvider.java [hunk 3]
```diff
@@ -9,14 +9,17 @@
 import greencity.enums.NotificationReceiverType;
 import greencity.exceptions.user.UserNotFoundException;
 import greencity.repository.NotificationTemplateRepository;
+import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 
 import javax.annotation.PostConstruct;
 
 @RequiredArgsConstructor
+@Getter
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/notification/AbstractNotificationProvider.java:18]: Why you added this?

File: service/src/main/java/greencity/service/notification/EmailService.java [hunk 4]
```diff
@@ -4,22 +4,29 @@
...
+import lombok.Getter;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.Objects;
 
+import static greencity.enums.NotificationReceiverType.EMAIL;
+
 @Service
+@Getter
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/notification/EmailService.java]: Why you added this?

File: service/src/main/java/greencity/ubstelegrambot/TelegramService.java [hunk 5]
```diff
@@ -15,19 +17,23 @@
...
 import java.util.Objects;
 
+import static greencity.enums.NotificationReceiverType.MOBILE;
+
 @Service
 @Slf4j
 public class TelegramService extends AbstractNotificationProvider {
     private final UBSTelegramBot ubsTelegramBot;
 
+    private static final NotificationReceiverType notificationType = MOBILE;
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/ubstelegrambot/TelegramService.java:27]:
```suggestion
    private static final NotificationReceiverType mobileNotificationType = MOBILE;
```

File: service/src/main/java/greencity/ubsviberbot/ViberServiceImpl.java [hunk 6]
```diff
@@ -39,6 +42,8 @@ public class ViberServiceImpl extends AbstractNotificationProvider implements Vi
     @Value("${greencity.bots.viber-bot-url}")
     private String viberBotUrl;
 
+    private static final NotificationReceiverType notificationType = MOBILE;
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/ubsviberbot/ViberServiceImpl.java:45]:
```suggestion
    private static final NotificationReceiverType mobileNotificationType = MOBILE;
```

File: service/pom.xml (hunk 2 above)
Comment (VadymMakitra) [inline service/pom.xml]:
```suggestion
            <version>${telegrambot.version}</version>
```

File: service/src/main/java/greencity/service/notification/AbstractNotificationProvider.java (hunk 3 above)
Comment (MarkAndzh) [reply service/src/main/java/greencity/service/notification/AbstractNotificationProvider.java:18]: I added it to call provider.getNotificationType() to pass notification type of provider into sendNotification function

**Pull request merged**